solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -45,10 +49,501 @@ describe("Bridge - Moving funds", () => {
+                await createSnapshot()
+
+                await bridge.setWalletMainUtxo(
+                  ecdsaWalletTestData.pubKeyHash160,
+                  mainUtxo
+                )
+              })
+
+              after(async () => {
+                await restoreSnapshot()
+              })
+
+              it("should revert", async () => {
+                // Changing the `txOutputValue` to a value other than `0` will
+                // make that scenario happen.
+                const corruptedMainUtxo = {
+                  ...mainUtxo,
+                  txOutputValue: 1,
+                }
+
+                await expect(
+                  // The last parameter doesn't matter in this scenario.
+                  bridge
+                    .connect(caller)
+                    .submitMovingFundsCommitment(
+                      ecdsaWalletTestData.pubKeyHash160,
+                      corruptedMainUtxo,
+                      walletMembersIDs,
+                      walletMemberIndex,
+                      []
+                    )
+                ).to.be.revertedWith("Invalid wallet main UTXO data")
+              })
+            })
+          })
+
+          context(
+            "when the caller is not a member of the source wallet",
+            () => {
+              const walletMembersIDs = [1, 2, 3, 4, 5]
+              const walletMemberIndex = 2
+
+              before(async () => {
+                await createSnapshot()
+
+                // That's the default behavior, but we just make it explicit.
+                walletRegistry.isWalletMember.returns(false)
+              })
+
+              after(async () => {
+                walletRegistry.isWalletMember.reset()
+
+                await restoreSnapshot()
+              })
+
+              it("should revert", async () => {
+                await expect(
+                  // The last parameter doesn't matter in this scenario.
+                  bridge
+                    .connect(thirdParty)
+                    .submitMovingFundsCommitment(
+                      ecdsaWalletTestData.pubKeyHash160,
+                      NO_MAIN_UTXO,
+                      walletMembersIDs,
+                      walletMemberIndex,
+                      []
+                    )
+                ).to.be.revertedWith(
+                  "Caller is not a member of the source wallet"
+                )
+              })
+            }
+          )
+        })
+
+        context("when the commitment was already submitted", () => {
+          before(async () => {
+            await createSnapshot()
+
+            await bridge.setWallet(ecdsaWalletTestData.pubKeyHash160, {
+              ...walletDraft,
+              state: walletState.MovingFunds,
+              // Set a non-zero commitment to make this scenario work.
+              movingFundsTargetWalletsCommitmentHash:
+                "0x959e95e0bd83d34878f77ead61cb4e955bf5e3bdc9e16cdfbd51c4c20ab7e6b4",
```

Comment:
We could instead just call `submitMovingFundsCommitment` in the `before`. I think calling it is *cleaner* than manipulating stored data.

Reply:
It's done that way because `submitMovingFundsCommitment` requires a specific state first and that will make the setup more complicated. Those are unit tests that should work in a specific initial state, and I think we should achieve that state in the most straightforward way since it is not the thing that matters here.